ir/src/transition_constraints/mod.rs

```diff
@@ -0,0 +1,64 @@
+use crate::error::SemanticError;
+
```

Comment:
minor nit: I don't think we need an extra line between these imports.

Reply:
Fixed!